jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/PaginatorTests.java

```diff
@@ -1,106 +1,148 @@
                     .map(String::valueOf)
                     .collect(Collectors.joining(","));
-    private static final int TOTAL = 50;
+    private static final int STEP = 100;
+    private static final int PAGE_SIZE = 10;
+    private static final int LENGTH = STEP * PAGE_SIZE - new Random().nextInt(STEP);
+    private static final String RANGE_PATTERN = "%d - %d / %d";
 
     @BeforeMethod
     public void before() {
-        getDriver().manage().window().setSize(new Dimension(1920, 1080));
-
-        shouldBeLoggedIn();
-        angularPage.shouldBeOpened();
-
-        listLength.setValue(String.valueOf(TOTAL));
-        pageSizeOptions.setValue(OPTIONS);
+        paginatorPage.open();
+        waitCondition(() -> paginatorPage.isOpened());
+        paginatorPage.checkOpened();
     }
 
-    @Test
+    @Test(description = "The test checks item per page label")
     public void labelPaginationTest() {
-        paginator.has().label("Items per page:");
+        paginatorConfigurable.has().itemPerPageLabel("Items per page:");
     }
 
-    @Test
+    @Test(description = "The test checks length and pageIndex for paginator")
     public void basicPaginatorTest() {
-        final int STEP = 10;
-        paginator.select(STEP);
-
-        paginator.is().range(1, STEP, TOTAL);
-        paginator.is().previousDisabled();
-        paginator.is().nextEnabled();
-        paginator.next();
-
-        for (int i = STEP + 1; i < TOTAL - STEP + 1; i += STEP) {
-            paginator.is().range(i, i + STEP - 1, TOTAL);
-            paginator.is().previousEnabled();
-            paginator.is().nextEnabled();
-            paginator.next();
+        waitCondition(() -> listLengthInput.isVisible());
+        listLengthInput.setValue(String.valueOf(LENGTH));
+        paginatorConfigurable.select(STEP);
+
+        //Go through each page sequentially:
+        for (int pageIndex = 0; pageIndex < PAGE_SIZE - 1; pageIndex++) {
+            final String rangeLabel = format(RANGE_PATTERN, pageIndex * STEP + 1, Math.min(pageIndex * STEP + STEP, LENGTH), LENGTH);
+
+            paginatorConfigurable.has().pageIndexCurrent(pageIndex)
+                    .and().has().totalNumberOfItems(LENGTH)
+                    .and().has().rangeLabel(rangeLabel);
+            paginatorConfigurable.nextPage();
         }
 
-        paginator.is().range(TOTAL - STEP + 1, TOTAL, TOTAL);
-        paginator.is().previousEnabled();
-        paginator.is().nextDisabled();
-        paginator.previous();
+        //Go through each page backwards
+        for (int pageIndex = PAGE_SIZE - 1; pageIndex > 0; pageIndex--) {
+            final String rangeLabel = format(RANGE_PATTERN, pageIndex * STEP + 1, Math.min(pageIndex * STEP + STEP, LENGTH), LENGTH);
 
-        for (int i = TOTAL - 2 * STEP + 1; i > 1; i -= STEP) {
-            paginator.is().range(i, i + STEP - 1, TOTAL);
-            paginator.is().previousEnabled();
-            paginator.is().nextEnabled();
```

Comment:
Perhaps revert previousEnabled() and nextEnabled() asserts as in the original test